src/Analytics/Adapter.php

```diff
@@ -62,4 +64,119 @@ public function disable(): void
      */
     abstract public function createEvent(Event $event): bool;
 
+    /**
+     * Call
+     *
+     * Make an API call
+     *
+     * @param string $method
+     * @param string $path
+     * @param array $params
+     * @param array $headers
+     * @return array|string
+     * @throws Exception
+     */
+    public function call($method, $path = '', $headers = array(), array $params = array())
```

Comment:
I left review regarding this method in this PR https://github.com/utopia-php/analytics/pull/20